cyclopts/core.py

```diff
@@ -1282,11 +1282,25 @@ def __call__(
                 if backend == "asyncio":
                     import asyncio
 
-                    return asyncio.run(command(*bound.args, **bound.kwargs))
+                    try:
+                        # Check if we're already in an async context
+                        asyncio.get_running_loop()
+                        # We're in an async context, return the coroutine for the caller to await
+                        return command(*bound.args, **bound.kwargs)
+                    except RuntimeError:
+                        # No running loop, create one
+                        return asyncio.run(command(*bound.args, **bound.kwargs))
```

Comment:
```suggestion
                    try:
                        # Check if we're already in an async context
                        asyncio.get_running_loop()
                    except RuntimeError:
                        # No running loop, create one
                        return asyncio.run(command(*bound.args, **bound.kwargs))
                    else:
                        # We're in an async context, return the coroutine for the caller to await
                        return command(*bound.args, **bound.kwargs)
```

to avoid swallowing any unrelated (also unexpected, I give it to you) `RuntimeError`s from instantiation of the coroutine